packages/flutter_tools/lib/src/commands/analyze_once.dart

```diff
@@ -49,34 +50,48 @@ class AnalyzeOnce extends AnalyzeBase {
     final bool currentPackage = argResults['current-package'] && (argResults.wasParsed('current-package') || dartFiles.isEmpty);
```

Comment:
I think we should remove `--current-directory` in this patch. This would simplify the logic here and is consistent with the long term plan in https://github.com/flutter/flutter/issues/7274.